go/ngql/types.go

```diff
@@ -748,3 +750,23 @@ func (it *mapIteratorForKeys) Next() (k, v types.Value) {
 	it.idx++
 	return
 }
+
+type setFirstIterator struct {
+	s types.Set
+}
+
+func (it *setFirstIterator) Next() types.Value {
+	return it.s.First()
```

Comment:
I initially kept track of done but turns out that we never hit that since we already handle the end using count.